pywbem_mock/_dmtf_schema.py

```diff
@@ -0,0 +1,392 @@
+        """
+        first = [True]  # List allows setting variable from inner function
+
+        def if_first(msg):
+            """
+            Inner method prints msg if var 'first' is False and verbose is
+            True. This allows displaying steps that are actually executed.
+            """
+            if first:
+                if self.verbose:
+                    print("")
+            first[0] = False
+            if self.verbose:
+                print(msg)
+
+        if not os.path.isdir(self.schema_dir):
+            if_first("Creating directory for CIM Schema archive: %s" %
+                     self.schema_dir)
+            os.mkdir(self.schema_dir)
+
+        if not os.path.isfile(self._mof_zip_filename):
+            if_first("Downloading CIM Schema archive from: %s" %
+                     self._mof_zip_url)
+
+            try:
+                ufo = urlopen(self._mof_zip_url)
+            except IOError as ie:
+                os.rmdir(self.schema_dir)
+                raise ValueError('Schema url %s not found. Exception: %s' %
+                                 (self._mof_zip_url, ie))
+
+            with open(self._mof_zip_filename, 'wb') as fp:
+                for data in ufo:
+                    fp.write(data)
+
+        if not os.path.isdir(self.schema_mof_dir):
+            if_first("Creating directory for CIM Schema MOF files: %s" %
+                     self.schema_mof_dir)
+            os.mkdir(self.schema_mof_dir)
+        if not os.path.isfile(self._schema_mof_filename):
+            if_first("Unpacking CIM Schema archive: %s" %
+                     self._mof_zip_filename)
+
+            zfp = None
+            try:
+                zfp = ZipFile(self._mof_zip_filename, 'r')
+                nlist = zfp.namelist()
+                for file_ in nlist:
+                    dfile = os.path.join(self.schema_mof_dir, file_)
+                    if dfile[-1] == '/':
+                        if not os.path.exists(dfile):
+                            os.mkdir(dfile)
+                    else:
+                        with open(dfile, 'w+b') as dfp:
+                            dfp.write(zfp.read(file_))
+            finally:
+                if zfp:
+                    zfp.close()
+
+    def build_schema_mof(self, schema_classes):
+        """
+        Build a mof file that includes the pragmas for the DMTF schema CIM
+        classes defined in 'schema_classes' using the DMTF CIM schema defined by
+        the directory 'schema_dir' and the schema version defined by
+        'schema_ver'.
+
+        It builds a compilable mof string in the form:
+
+            pragma locale ("en_US")
+            pragma include ("System/CIM_ComputerSystem.mof")
+
+        Parameters:
+
+          schema_classes (:term:`py:list` of :term:`string`):
+            These must be classes in the define DMTF schema and can be just
+            a list of the leaf classes required for a working class repository.
```

Comment:
I suggest to expand a little on "just a list of leaf classes" by mentioning the automatic dependency resolution. Maybe in the method description, not in the parameter description.